command/agentproxyshared/cache/lease_cache.go

```diff
@@ -263,19 +318,52 @@ func (c *LeaseCache) Send(ctx context.Context, req *SendRequest) (*SendResponse,
 		case <-inflight.ch:
 		}
 	} else {
-		inflight = newInflightRequest()
+		if inflight == nil {
```

Comment:
What's the purpose of this `if`?  When do we want to re-use an existing `inflight`?

Reply:
It's so that we have one inflight irrespective of if it's a dynamic or static request, essentially. It prevents double `defer close(inflight.ch)` and `defer inflight.remaining.Dec()` etc.

Reply:
How could we have two?  How could the same request give rise to a cache hit for both keys?

Reply:
It won't result in an inflight cache hit for both keys, but we take the `else` branch if we get a cache miss. We also don't know at this stage if it's a cache hit for either the static secret ID or the dynamic secret ID

Reply:
To put it another way, it's likely for most requests to result in a cache miss for both of the inflight cache IDs

Reply:
This particular `else` could never see a non-nil inflight though, right?  For the second case, the static if/else, might it be clearer to wrap the entire thing in an `if inflight == nil` instead?  If we have already found the dynamic cache entry, there's no point in doing idLockStaticSecret.Lock and consulting the cache again, is there?

Reply:
We still need to set both of the IDs in the cache (we don't know what kind of request it is at this stage) and unlock the inflight cache lock, so that behaviour can't be conditional. I might be misunderstanding, though?

We only have one request inflight (the stuff protected by the `if inflight == nil`) but we need to lock the IDs as it could be dynamic or static (or neither)

Reply:
Ah, I'm the one who was missing something, I agree.  Though I still feel the first `if inflight == nil` is a no-op, the one this comment is tied to.